Applications/FileIO/Legacy/OGSIOVer4.cpp

```diff
@@ -475,32 +475,39 @@ bool readGLIFileV4(const std::string& fname,
     else
         INFO("GeoLib::readGLIFile(): tag #POLYLINE not found.");
 
+    if (!ply_vec->empty())
+        geo.addPolylineVec(
+            std::move(ply_vec), unique_name,
+            std::move(ply_names));  // KR: insert into GEOObjects if not empty
+
+    // copy of ply_names
+    std::map<std::string, std::size_t> ply_names_copy{
+        geo.getPolylineVecObj(unique_name)->getNameIDMapBegin(),
+        geo.getPolylineVecObj(unique_name)->getNameIDMapEnd()};
+
     auto sfc_vec = std::make_unique<std::vector<GeoLib::Surface*>>();
     auto sfc_names = std::make_unique<std::map<std::string, std::size_t>>();
     if (tag.find("#SURFACE") != std::string::npos && in)
     {
         INFO("GeoLib::readGLIFile(): read surfaces from stream.");
+
         tag = readSurfaces(in,
                            *sfc_vec,
                            *sfc_names,
-                           *ply_vec,
-                           *ply_names,
+                           *geo.getPolylineVec(unique_name),
+                           ply_names_copy,
```

Comment:
Since `ply_names` is a `auto_ptr` and is given to the `GEOObject` object `geo` it is not usable anymore.  For this reason a copy is necessary.